### DISCUSSION TITLE
Package meta file corruption after migration to v5

### DISCUSSION THREAD

**Discussion by OverLord561** [verdaccio/verdaccio#4869] [Q&A] [answered]
**Your Environment**
  
 * **verdaccio version**: 5.32.2
 * **previous verdaccio version**: 4.12
 * **node version** 20
 * **package manager**: [pnpm@8]
 * **os**: [linux]
 
**Describe the bug**
We tried to upgrade from v4 to v5.32.2. The problem is that meta file is truncated. The SSR relies on **renderingService** meta field to determine what is the entry point for rendering but that metadata is simply missed
![image](https://github.com/user-attachments/assets/22f6c07d-d79b-43fd-a06f-55689afb6dc6)

We tried to upgrade the libs which work with verdaccio but no luck.
![image](https://github.com/user-attachments/assets/500cdfa2-26c8-41fa-a747-39152f8ce878)

Would be grateful for any ideas.
Thanks in advance!

**Comment by OverLord561** [comment]
Breaking change https://github.com/verdaccio/verdaccio/releases/tag/v5.15.0

**Comment by OverLord561** [comment, marked as answer]
This issue has nothing to do with the Verdaccio itself - that is the client.

Once you use something like @pnpm/client you might need **fullMetadata: true** during client creation to achieve old behavior when metadata is not optimized.

**Reply by OverLord561** [reply]
While the new behavior is
![image](https://github.com/user-attachments/assets/d3f69f24-7aad-40be-98c9-a93355986700)

**Comment by mbtools** [comment]
You got it. Since 5.15, verdaccio behaves like the npm registry and returns the abbreviated manifest if the client asks for it.
